huggingface_pipelines/pipeline_config.py

```diff
@@ -0,0 +1,179 @@
+from abc import ABC
+from dataclasses import dataclass, replace
+from typing import List
+
+
+@dataclass
+class PipelineConfig(ABC):
+    """
+    Abstract base class for pipeline configurations.
+
+    Attributes:
+        dataset_name (str): The name of the HuggingFace dataset to be used.
+        dataset_split (str): The dataset split to be used (e.g., 'train', 'test', 'validation').
+        batch_size (int): The batch size to be used for processing.
+        device (str): The device to use for inference (e.g., 'cpu', 'cuda').
+        pipeline_type (str): The type of pipeline (e.g., 'text', 'audio').
+        num_shards (int): The number of shards to split the dataset into. Defaults to 1.
+        shard_id (int): The ID of the shard to process. Defaults to 0.
+        cache_to_arrow (bool): Whether to cache results to Arrow format. Defaults to False.
+        output_file_name (str): The base name of the file where results will be saved. Defaults to "results".
+        take (int): The number of batches to take for processing. Defaults to -1 (process all).
+
+    """
+    dataset_name: str
+    columns: List[str]
+    dataset_split: str = "test"
+    batch_size: int = 5
+    device: str = "cpu"
+    num_shards: int = 1
+    shard_id: int = 0
+    cache_to_arrow: bool = False
+    output_file_name: str = "results"
```

Comment:
this should be a directory rather than a file (for bigger datasets), `output_dir`.
And  output_file_name should not be mandatory (and especially without default name)

Reply:
Done will update in the new pr changes